openadapt/privacy/providers/private_ai.py

```diff
@@ -0,0 +1,184 @@
+            file_data = file_data.decode("ascii")
+
+        payload = {
+            "file": {"data": file_data, "content_type": file_type},
+            "entity_detection": {"accuracy": "high", "return_entity": True},
+            "pdf_options": {"density": 150, "max_resolution": 2000},
+            "audio_options": {"bleep_start_padding": 0, "bleep_end_padding": 0},
+        }
+
+        headers = {
+            "Content-Type": "application/json",
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(url, json=payload, headers=headers)
+        if response is None:
+            raise ValueError("Private AI request returned None")
+        response = response.json()
+        logger.debug(response)
+        if type(response) is dict and "detail" in response.keys():
+            raise ValueError(response.get("detail"))
+
+        redact_file_path = os.path.join(file_dir, f"redacted-{file_name}")
+
+        # Write to file
+        with open(redact_file_path, "wb") as redacted_file:
+            processed_file = response.get("processed_file").encode("ascii")
+            processed_file = base64.b64decode(processed_file, validate=True)
+            redacted_file.write(processed_file)
+
+        # return PIL_IMAGE type of redacted image
+        with open(redact_file_path, "rb") as file:
+            redacted_file_data = file.read()
+
+        redact_pil_image_data = Image.open(BytesIO(redacted_file_data))
+
+        os.remove(temp_image_path)
+        os.remove(redact_file_path)
+
+        return redact_pil_image_data
+
+    def scrub_pdf(self, path_to_pdf: str) -> str:
+        """Scrub the PDF of all PII/PHI.
+
+        Args:
+            path_to_pdf (str): Path to the PDF to be scrubbed
+
+        Returns:
+            str: Path to the scrubbed PDF
+        """
+        url = "https://api.private-ai.com/deid/v3/process/files/base64"
+
+        file_type = "application/pdf"
+
+        # Read from file
+        with open(path_to_pdf, "rb") as b64_file:
+            file_data = base64.b64encode(b64_file.read())
+            file_data = file_data.decode("ascii")
+
+        payload = {
+            "file": {"data": file_data, "content_type": file_type},
+            "entity_detection": {"accuracy": "high", "return_entity": True},
+            "pdf_options": {"density": 150, "max_resolution": 2000},
+            "audio_options": {"bleep_start_padding": 0, "bleep_end_padding": 0},
+        }
+
+        headers = {
+            "Content-Type": "application/json",
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(url, json=payload, headers=headers)
+        if response is None:
+            raise ValueError("Private AI request returned None")
+        response = response.json()
+        if type(response) is dict and "details" in response.keys():
+            raise ValueError(response.get("detail"))
+        logger.debug(response.get("entities"))
```

Comment:
Please use f-strings to indicate what is being logged, i.e.

```
        logger.debug(f"{response.get('entities')=}")
```

Reply:
Addressed in: a37ee99ecc6ce1bed52b3cc3918c93495c06b785